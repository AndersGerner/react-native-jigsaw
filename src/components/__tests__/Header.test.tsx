import * as React from "react";
import renderer from "react-test-renderer";
import Header from "../Header";

it("renders container", () => {
  const tree = renderer
    .create(
      <Header
        onPress={() => {}}
        buttonText="Test"
        icon="check"
        title="Test Title"
        titleColor="black"
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
