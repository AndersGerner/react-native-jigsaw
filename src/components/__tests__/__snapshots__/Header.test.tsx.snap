// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders container 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          undefined,
          Object {
            "color": "black",
            "flex": 1,
          },
        ]
      }
    >
      Test Title
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "marginLeft": 12,
          "opacity": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 22,
                },
                Object {
                  "color": "rgba(165, 173, 183, 1)",
                  "marginRight": 8,
                },
              ]
            }
          >
            Test
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(234, 237, 242, 1)",
        },
        Object {
          "marginTop": 16,
        },
        Object {
          "height": 0.5,
        },
      ]
    }
  />
</View>
`;
